app/src/main/java/com/cops/iitbhu/previewer/MainActivity.kt

```diff
@@ -1,71 +1,30 @@
 package com.cops.iitbhu.previewer
 
+import android.content.Intent
 import android.os.Bundle
 import android.view.View
-import android.widget.Toast
-import androidx.activity.result.contract.ActivityResultContracts
 import androidx.appcompat.app.AppCompatActivity
-import androidx.lifecycle.lifecycleScope
 import com.cops.iitbhu.previewer.databinding.ActivityMainBinding
-import com.cops.iitbhu.previewer.lib.Previewer
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityMainBinding
 
-//    private val activityResultLauncherForVideoUri =
-//        registerForActivityResult(ActivityResultContracts.GetContent()) { videoUri: Uri ->
-//            Previewer.setVideoThumbnailFromUri(videoUri, binding.imgThumbnail)
-//        }
-
-    private val getUriOfPdf =
-        registerForActivityResult(ActivityResultContracts.OpenDocument()) { uri ->
-
-            if (uri != null) {
-                val bitmap = Previewer.generateBitmapFromPdf(uri)
-                if (bitmap != null)
-                    binding.img.setImageBitmap(bitmap)
-                else
-                    Toast.makeText(this, "Unable to generate bitmap!", Toast.LENGTH_SHORT).show()
-            }
-        }
-
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
 
-        binding.selectFile.setOnClickListener(View.OnClickListener {
-            getUriOfPdf.launch(arrayOf("application/pdf"))
+        binding.pdf.setOnClickListener(View.OnClickListener {
+            startActivity(Intent(this@MainActivity, BitmapPDF::class.java))
```

Comment:
Replace `this@MainActivity` with just `this` in all three click listeners.

Reply:
done